src/postgres/commands/configurePostgresFirewall.ts

```diff
@@ -7,43 +7,74 @@ import PostgreSQLManagementClient from "azure-arm-postgresql";
 import { FirewallRule } from "azure-arm-postgresql/lib/models";
 import * as publicIp from 'public-ip';
 import * as vscode from 'vscode';
-import { createAzureClient, DialogResponses, IActionContext } from "vscode-azureextensionui";
+import { createAzureClient, DialogResponses, IActionContext, IAzureQuickPickItem } from "vscode-azureextensionui";
 import { ext } from "../../extensionVariables";
+import { IPostgresWizardContext } from "../../tree/PostgresAccountWizard/IPostgresWizardContext";
 import { azureUtils } from "../../utils/azureUtils";
 import { localize } from "../../utils/localize";
 import { nonNullProp } from "../../utils/nonNull";
 import { PostgresServerTreeItem } from "../tree/PostgresServerTreeItem";
 
-export async function configurePostgresFirewall(context: IActionContext, treeItem?: PostgresServerTreeItem): Promise<void> {
-    if (!treeItem) {
+export async function configurePostgresFirewall(context: IActionContext | IPostgresWizardContext, treeItem?: PostgresServerTreeItem, createMode?: boolean): Promise<string> {
+    const ip: string = await publicIp.v4();
+    if (!treeItem && createMode) {
```

Comment:
All of the stuff inside of this `if` case should be inside `PostgresServerFirewallStep.ts` instead. It's only related to the prompt step, not running the command. There should be other ways to share code if necessary